ledger/eval/eval.go

```diff
@@ -1160,12 +1167,27 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 	return nil
 }
 
+func (cs *roundCowState) takeFee(tx *transactions.Transaction, senderRewards *basics.MicroAlgos, ep *logic.EvalParams) error {
+	err := cs.Move(tx.Sender, ep.Specials.FeeSink, tx.Fee, senderRewards, nil)
+	if err != nil {
+		return err
+	}
+	// transactions from FeeSink should be exceedingly rare. But we can't count
+	// them in feesCollected because there are no net algos added to the Sink
+	if tx.Sender == ep.Specials.FeeSink {
```

Comment:
nonetheless, if I'm proposing blocks, is it my fault that I only have a bunch of FeeSink transactions?  In that situation the proposer gest nothing right?

Reply:
It's possible we could go even further than this PR currently proposes and ban _all_ FeeSink transactions. (We only ban pays now.). I don't see any real problems doing that, actually.

Reply:
I think that makes more sense than marking them as paying 0 mining rewards.

Reply:
Yes, we're going to ban all `FeeSink` transactions.